doc/userguide/lua/libs/ja3.rst

```diff
@@ -0,0 +1,90 @@
+JA3
+---
+
+JA3 details are exposes to Lua scripts with the
+``suricata.ja3`` library, For example::
+
+  local ja3 = require("suricata.ja3")
+
+If you want to use ja3, you can either set suricata.yaml option
+``app-layer.protocols.tls.ja3-fingerprints`` to true,
+or specify it in the ``init`` function of your lua script
+by calling ``ja3.enable_ja3()``::
+
+  function init (args)
+    ja3.enable_ja3()
+    return {}
+  end
+
+``ja3.enable_ja3()`` will not enable ja3 if they are explicitly
+disabled, so you should add ``requires: feature ja3;`` to your rule.
+
+For use in rule matching, the rule may **hook** into a TLS or QUIC
+transaction state if you want to match on only one of these protocols.
+Or you should use need ``ja3`` or ``ja3s`` in your init script::
```

Comment:
Could you clarify what hook means here? Are these rule hooks? If so, I suggest adding a reference to that. If something else, maybe we need rewording or a longer explanation? (or a pointer to where to find more info)

Reply:
> Could you clarify what hook means here?

Not sure I can do it right, it comes from @jasonish doc about lua dns.

I am removing this as `needs["ja3s"] = true` looks more appropriate for ja3 lib